code/modules/maps/ruins.dm

```diff
@@ -17,12 +17,12 @@ var/list/banned_ruin_ids = list()
 	var/remaining = budget
 
 	for(var/datum/map_template/ruin/ruin in potentialRuins)
-		if(ruin.template_flags & TEMPLATE_FLAG_SPAWN_GUARANTEED)
+		if((ruin.spawns_in_current_sector()) && (ruin.template_flags & TEMPLATE_FLAG_SPAWN_GUARANTEED))
```

Comment:
```suggestion
		if(HAS_FLAG(ruin.template_flags, TEMPLATE_FLAG_SPAWN_GUARANTEED) && (ruin.spawns_in_current_sector()))
```
Since we're at it, reformat it to use the flag macro, and since DM uses lazy evaluation put the less expensive check first, so if it doesn't have the flag, we avoid wasting time with the function call